subsys/mgmt/mcumgr/zephyr_grp/basic_mgmt.c

```diff
@@ -33,6 +36,89 @@ int storage_erase(void)
 	return rc;
 }
 
+
+#if defined(CONFIG_MCUMGR_GRP_BASIC_CMD_IMAGE_LIST)
+static int image_status(int id, char *buffer, ssize_t len)
+{
+	const struct flash_area *fa;
+	struct image_header hdr;
+	struct boot_swap_state bss;
+	int rc = flash_area_open(id, &fa);
+
+	if (rc < 0) {
+		LOG_ERR("Flash area %d open failed", id);
+		return -1;
+	}
+
+	rc = flash_area_read(fa, 0, &hdr, sizeof(hdr));
+	if (rc < 0) {
+		flash_area_close(fa);
+		LOG_ERR("Flash area %d read failed", id);
+		return -1;
+	}
+
+	rc = boot_read_swap_state(fa, &bss);
+	flash_area_close(fa);
+	if (rc < 0) {
+		LOG_ERR("Boot swap state %d read failed", id);
+		return -1;
+	}
+	if (hdr.ih_magic == IMAGE_MAGIC) {
+		snprintf(buffer, len, "ver=%d.%d.%d.%d%s", hdr.ih_ver.iv_major,
+			 hdr.ih_ver.iv_minor, hdr.ih_ver.iv_revision,
+			 hdr.ih_ver.iv_build_num,
+			 bss.copy_done == BOOT_FLAG_SET ? ",copy_done" : "");
+		return 1;
+	}
+	/* Not an application, not an error */
+	return 0;
+}
+
+/* Only defined images are probed */
+#define IMAGE_LIST_ITEM(n, node_label)					\
+	COND_CODE_1(DT_HAS_FIXED_PARTITION_LABEL(node_label),		\
+		({							\
+			.num = n,					\
+			.fa_id = FLASH_AREA_ID(node_label),		\
+		},), ())
+
+static int image_list(struct mgmt_ctxt *ctxt)
+{
+	static const struct {
+		int num; int fa_id;
+	} images[] = {
+		IMAGE_LIST_ITEM(1, image_0)
+		IMAGE_LIST_ITEM(2, image_1)
+		IMAGE_LIST_ITEM(3, image_2)
+		IMAGE_LIST_ITEM(4, image_3)
+	};
+	CborError cbor_err = 0;
+	char buffer[64];	/* Buffer should fit version and flags */
+
+	for (int i = 0; cbor_err == 0 && i < ARRAY_SIZE(images); i++) {
+		if (image_status(images[i].num, buffer, sizeof(buffer)) > 0) {
+			cbor_err |= cbor_encode_int(&ctxt->encoder, images[i].num);
```

Comment:
Can we use {image, slot} coordinate indexes instead:
{0,0}, {0,1}
{1,0) {1,1}
Or not really as we are using MAP

Reply:
We can, but it does not really make sense. For example, when you select image number 2 to upload;
`mcumgr ... image upload -e -n 2 blah.bin`
it would be translated to [1,0], and this is counter intuitive, to see:
```
{0,0} v1.7.0
{0,1} v1.7.1
{1,0} v3.4.5,upload_in_progress
{1,1}
```
Rather than:
```
1: v1.7.0
2: v1.7.1
3: v3.4.5,upload_in_progress
4:
```

we can also do the double mapping
```
1: {0,0} v1.7.0
2: {0,1} v1.7.1
3: {1,0} v3.4.5,upload_in_progress
4: {1,1}
```

This is more about how you want to read it than what to return. When code requesting update adds, to the update frame,  `image=2` it will be easier for it to search for `2:`, in response, rather than translate it to `{1,0}`, and then search; the statuses are returned for all known images so the `image=1` may also be used as index of status (less 1).